drkafka/src/main/java/com/pinterest/doctorkafka/modules/context/KafkaClusterContext.java

```diff
@@ -0,0 +1,81 @@
+package com.pinterest.doctorkafka.modules.context;
+
+import com.pinterest.doctorkafka.DoctorKafkaActionReporter;
+import com.pinterest.doctorkafka.KafkaCluster;
+import com.pinterest.doctorkafka.util.ZookeeperClient;
+
+import kafka.cluster.Broker;
+import kafka.utils.ZkUtils;
+import org.apache.kafka.common.PartitionInfo;
+
+import java.util.List;
+
+public class KafkaClusterContext extends ClusterContext {
+  private KafkaCluster kafkaCluster;
+  private List<PartitionInfo> underReplicatedPartitions;
+  private List<Broker> noStatsBrokers;
+  private DoctorKafkaActionReporter actionReporter;
+  private ZkUtils zkUtils;
+  private ZookeeperClient zookeeperClient;
```

Comment:
This was confusing during refactoring too. We should either 1) provide a generic dependency for that is general enough for all plugins to utilize, which would need some consolidation of common use cases, or 2) have plugins create the instances inside their own scope to reduce pollution of the context. Since this current PR is only porting original code, I left the dependency here to minimize the number of modifications.